pkg/apis/config/image_policies.go

```diff
@@ -76,10 +118,69 @@ func parseEntry(entry string, out interface{}) error {
 }
 
 // GetMatchingPolicies returns all matching Policies and their Authorities that
-// need to be matched for the given Image.
+// need to be matched for the given kind, version and labels (if provided) to then match the Image.
+// Returned map contains the name of the CIP as the key, and a normalized
+// ClusterImagePolicy for it.
+func (p *ImagePolicyConfig) GetMatchingPolicies(image string, kind, apiVersion string, labels map[string]string) (map[string]webhookcip.ClusterImagePolicy, error) {
+	if p == nil {
+		return nil, errors.New("config is nil")
+	}
+
+	var lastError error
+	ret := make(map[string]webhookcip.ClusterImagePolicy)
+
+	// TODO(vaikas): this is very inefficient, we should have a better
+	// way to go from image to Authorities, but just seeing if this is even
+	// workable so fine for now.
+	for k, v := range p.Policies {
+		if len(v.Match) > 0 {
+			foundMatch := false
+			resourceGroupVersion := kindResourceMap[kind]
+			for _, matchResource := range v.Match {
+				// TODO: Check version
+				if matchResource.Resource == resourceGroupVersion.Resource && matchResource.Version == resourceGroupVersion.Version && matchResource.Group == resourceGroupVersion.Group {
+					if matchResource.ResourceSelector != nil {
+						selector, err := metav1.LabelSelectorAsSelector(matchResource.ResourceSelector)
+						if err != nil {
+							return nil, errors.New("policy with wrong match label selector")
+						}
+						if !selector.Matches(metalabels.Set(labels)) {
+							continue
+						}
+						// We found a resource type that matches the provided labels
+						foundMatch = true
+						break
+					} else {
+						// We found a resource that matches the available name, version and group
+						foundMatch = true
+						break
+					}
+				}
+			}
+			if !foundMatch {
+				// We didn't find any match with the current resource types, so we continue looking for policies
+				continue
+			}
+		}
+
+		for _, pattern := range v.Images {
+			if pattern.Glob != "" {
+				if matched, err := glob.Match(pattern.Glob, image); err != nil {
+					lastError = err
+				} else if matched {
+					ret[k] = v
+				}
+			}
+		}
+	}
+	return ret, lastError
+}
+
+// GetMatchingPolicies returns all matching Policies and their Authorities that
+// need to be matched the Image.
 // Returned map contains the name of the CIP as the key, and a normalized
 // ClusterImagePolicy for it.
-func (p *ImagePolicyConfig) GetMatchingPolicies(image string) (map[string]webhookcip.ClusterImagePolicy, error) {
+func (p *ImagePolicyConfig) GetMatchingPoliciesByImage(image string) (map[string]webhookcip.ClusterImagePolicy, error) {
```

Comment:
Is this now only being used by tester?

Reply:
yes, that is correct. I wanted to leave it, but I am happy to polish this code.

Reply:
I guess my only thinking is that if they then start diverging. Maybe just get rid of it now, and pass in fields in there that don't restrict anything?

Reply:
Sounds good to me!